syncer/syncer.go

```diff
@@ -1740,10 +1738,23 @@ type eventContext struct {
 
 // TODO: Further split into smaller functions and group common arguments into a context struct.
 func (s *Syncer) handleRotateEvent(ev *replication.RotateEvent, ec eventContext) error {
-	if ec.header.Timestamp == 0 || ec.header.LogPos == 0 { // fake rotate event
-		if string(ev.NextLogName) <= ec.lastLocation.Position.Name {
+	failpoint.Inject("MakeFakeRotateEvent", func(val failpoint.Value) {
+		ec.header.LogPos = 0
+		ev.NextLogName = []byte(val.(string))
+		ec.tctx.L().Info("MakeFakeRotateEvent", zap.String("fake filen name", string(ev.NextLogName)))
+	})
+
+	if utils.IsFakeRotateEvent(ec.header) {
+		if fileName := string(ev.NextLogName); fileName <= ec.lastLocation.Position.Name {
```

Comment:
new binlog filename of new server may not <= old filename, 

mysqlbin.xxx vs bin.xxx

Reply:
i not sure about this, you can check this [commit](https://github.com/pingcap/dm/commit/b6a6e585b8cc523fa406758c5fb2ef86801f4c89)

Reply:
that commit is to handle this case:

When we restart synchronization from an early position, DM will see lots of oudated rotate event (we could determine if they are outdated by comparing the filename with global checkpoint). For some reason (I forgot) DM will update table checkpoint to that outdated position. Because the position is older than global checkpoint, DM will panic https://github.com/pingcap/dm/blob/18918ed3d4813a61a43cabf04e5e1fc28bfd5a43/syncer/checkpoint.go#L363

Reply:
`mysqlbin.999999` vs `mysqlbin.1000000` 🤔

Reply:
should use `CompareBinlogFileName`

Reply:
fixed in  https://github.com/pingcap/dm/pull/1848/commits/d8cac8507fb9f55642f5aafb0083ab9307099010